pages/agent/v3/signed_pipelines.md

```diff
@@ -0,0 +1,162 @@
+  <ul>
+    <li>It had to be installed on every agent instance, leading to more configuration.</li>
+    <li>It only supported symmetric HS256 signatures, meaning that every verifier could also sign uploads.</li>
+    <li>It couldn't sign <a href="/docs/pipelines/build-matrix">matrix steps</a>.</li>
+  </ul>
+  <p>This newer version of pipeline signing is built right into the agent and addresses all of these limitations. Being built into the agent, it's also easier to configure and use.</p>
+  <p>Many thanks to <a href="seek.com.au">SEEK</a>, who we collaborated with on the older version of the tool, and whose prior art has been instrumental in the development of this newer version.</p>
+</details>
+
+## Pipeline signatures
+
+Pipeline signatures establish that important aspects of steps haven't been changed since they were uploaded.
+
+The following fields are included in the signature for each step:
+
+- **Commands.**
+- **Environment variables defined in the pipeline YAML.** Environment variables set by the agent, hooks, or the user's shell are not signed and can override the environment a step's command is started with.
+- **Plugins and plugin configuration.**
+- **Matrix configuration.** The matrix configuration is signed as a whole rather than each individual matrix job. This means the signature is the same for each job in the matrix. When signatures are verified for matrix jobs, the agent double-checks that the job it received is a valid construction of the matrix and that the signature matches the matrix configuration.
+- **The step key.** If there is a step key present. [WIP]
+
+Additionally, the following aspects of the build for each job are included in the signature:
+
+- The pipeline's UUID [WIP]
+- The organization UUID [WIP]
+- The build number [WIP]
+- The repo the build is running for [WIP]
+- The commit the build is running for [WIP]
+
+## Enabling signed pipelines on your agents
+
+You'll need to configure your agents and update pipeline definitions to enable signed pipelines.
+
+### 1. Generate a key pair
+
+Behind the scenes, signed pipelines use [JSON Web Signing (JWS)](https://datatracker.ietf.org/doc/html/rfc7797) to generate signatures. You'll need to generate a [JSON Web Key Set (JWKS)](https://datatracker.ietf.org/doc/html/rfc7517) to sign and verify your pipelines with, then configure your agents to use those keys.
+
+Luckily, the agent has you covered! A JWKS generation tool is built into the agent, which you can use to generate a key pair for you. To use it, you'll need to [install the agent on your machine](/docs/agent/v3/installation), and then run:
+
+```bash
+buildkite-agent tool keygen --alg <algorithm> --key-id <key-id>
+```
+
+Replacing the following:
+
+- `<algorithm>` with the signing algorithm you want to use.
+- `<key-id>` with the key ID you want to use.
+
+For example, to generate an EdDSA key pair with a key ID of `my-key-id`, you'd run:
```

Comment:
I still think it would be good to add a link on "EdDSA"

Reply:
https://en.wikipedia.org/wiki/EdDSA should be fine i think, though we go into a bit more detail about EdDSA in particular a bit further down